flair/trainers/trainer.py

```diff
@@ -408,16 +539,6 @@ def train(
 
                 # process mini-batches
                 for batch_no, batch in enumerate(batch_loader):
-                    batch_kw = {
-                        "batch_no": batch_no,
-                        "batch": batch,
-                        "total_number_of_batches": len(batch_loader),
-                        "epoch": epoch,
-                    }
-
-                    # - TrainingBehaviorPlugin sets batch loss and number of samples to 0 (TODO: dispatch with only 1 customer)
-                    self.dispatch("before_training_batch", **batch_kw)
```

Comment:
Though none of the included plugins use this event, I think it might be helpful to keep it, as it is useful for extending the functionality of the train function. The same applies e.g. to `after_training_batch_step`.